src/main.py

```diff
@@ -71,9 +70,39 @@ def main():
                 bt.connect(BT_ADDR)
                 sleep(Follower.CYCLE_TIME)
 
-        leader = Leader()
-        while True:
-            leader.lead()
+        vehicle = Leader()
+
+    # This loop makes the vehicle move. If the vehicle sees an ARTag then
+    # it is a follower vehicle, otherwise it is a leader vehicle.
+    timer_set = False
+    start_time = time.time()
+    while True:
+        tag_visible = tag.detect()
+        if isinstance(vehicle, Leader):
```

Comment:
Turn this whole block into a bunch of chunks. It makes it really hard to read and follow the code without few blank lines to separate related code